front/lib/resources/space_resource.ts

```diff
@@ -345,6 +345,17 @@ export class SpaceResource extends BaseResource<SpaceModel> {
     }
 
     await this.update({ name: newName });
+    // For regular spaces that only have a single group, update
+    // the group's name too (see https://github.com/dust-tt/tasks/issues/1738)
+    const singleGroup = this.groups[0];
+    if (
+      singleGroup &&
+      this.groups.length === 1 &&
+      (this.kind === "regular" || this.kind === "public")
```

Comment:
```suggestion
      (this.isRegular() || this.isPublic())
```

Reply:
Ack, will update since the functions are already there and used.

But FTR I think it's better not to encapsulate straightforward single-clause conditions; makes the code harder to read IMO

I beleive we should start encapsulating at 2 clauses when there's reuse and a well-defined concise semantic

"Premature refactorization is the root of all evil in computer programming" as Jesus said ;)

Reply:
ofc, but IMO the field kind is not an internal concept, it's part of the public interface of groups